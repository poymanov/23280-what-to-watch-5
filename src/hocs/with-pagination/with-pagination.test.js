import React from "react";
import renderer from "react-test-renderer";
import withPagination from "./with-pagination";

const movie = {
  id: 1,
  name: `test`,
  posterImage: `test`,
  previewImage: `test`,
  backgroundImage: `test`,
  backgroundColor: `test`,
  videoLink: `test`,
  previewVideoLink: `test`,
  description: `test`,
  rating: 1,
  scoresCount: 1,
  director: `test`,
  starring: [`test`],
  runTime: 1,
  genre: `test`,
  released: 1,
  isFavorite: false,
};

const MockComponent = () => {
  return (<div />);
};

const MockComponentWrapped = withPagination(MockComponent);

it(`withPagination is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped movies={[movie]}>
      <React.Fragment />
    </MockComponentWrapped>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
