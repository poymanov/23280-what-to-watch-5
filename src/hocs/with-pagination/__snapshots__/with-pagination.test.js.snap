// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withPagination is rendered correctly 1`] = `<div />`;
